core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -532,6 +541,8 @@ public List<AcquiredRecords> acquire(
         RecordBatch firstBatch = fetchPartitionData.records.batches().iterator().next();
         lock.writeLock().lock();
         try {
+            // Update the latest fetch offset metadata for any future queries.
+            this.latestFetchOffsetMetadata = fetchOffsetMetadata;
```

Comment:
So in a fetch result of 10 batches we should store the last batch info, is that correct? If yes then shouldn't name of variable be appropriately defined?

Reply:
we store the last fetched offset's properties in `LogOffsetMetadata` class like the message offset, file position etc. Hence, naming it `latestFetchOffsetMetadata` makes more sense than `latestFetchBatchMetadata`. Wdyt?